src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/SerializableInfo.cs

```diff
@@ -0,0 +1,43 @@
+﻿//-----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//-----------------------------------------------------------------------------
+namespace System.Runtime.Serialization
+{
+    using System;
+    using System.Diagnostics;
+
+    public class SerializableInfo
+    {
+        public string TypeName { get; set; }
+        public string[] NonSerializedMembers { get; set; }
+
+        private SerializableInfo(string typeName, string[] nonSerializedMembers)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(typeName));
+            Debug.Assert(nonSerializedMembers != null);
+            TypeName = typeName;
+            NonSerializedMembers = nonSerializedMembers;
+        }
+
+        public static SerializableInfo Parse(string str)
+        {
+            if (string.IsNullOrEmpty(str))
+            {
+                throw new ArgumentNullException("str");
+            }
+            string[] parts = str.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Per my previous comments above, I think `SerializableInfo` can be removed altogether, but if there is a reason it needs to be kept around, consider caching the `new char[] { ';' }` in a static readonly field to avoid allocating the new array each time `Parse` is called.